### DISCUSSION TITLE
Merge back to mainline

### DISCUSSION THREAD

**Discussion by kyr0** [PCemOnMac/PCemV17macOS#14] [Q&A] [answered]
Hi folks, 

especially @kode54 @almeath -- how do you feel about merging back to mainline? I was very happy to read Sarahs' reply to my 2019 post. I'd love to take the effort and bring this all back in harmony with the mainline again and maybe even make official support for the Mac platform possible from V18 on?

Sarah did ask today: http://pcem-emulator.co.uk/phpBB3/viewtopic.php?p=15407&sid=6d0bcab4917dffacc0a5dd21cbbbf7e6#p15407

@kode54 I see light at the end of the tunnel for the arm64 code generator backend as well if we all work on this together :)

It might of course be a bit of effort to create the patches cause it might be quite a bunch of them now :))

But as far as I can see, we could come up with a few PR's maybe. Not sure yet. I think we should ask her what she'd prefer. https://github.com/sarah-walker-pcem/pcem/

What do you think?

Best,
Aron

**Comment by kode54** [comment]
I like the idea of merging back into mainline. And I do agree that with this increased platform coverage, and the greater ease of debugging arm64 on an actual computer compared to a mobile device, should make it a lot easier to hopefully fix the arm64 codegen to be in line with the x86_64 one.

**Comment by ghost** [comment]
Sorry, I somehow did not even see this discussion!

I'm also in favor of merging with mainstream.

**Comment by Shambler2** [comment]
Same here - happy to help with testing what platforms I have available (Win/Linux - x64 & BigSur ARM)

What's the best way to prepare a build ready for testing? Imagine it's a fork of this? Was going to attempt a basic comparison myself but wasn't sure if anyone else had already started

**Comment by almeath** [comment]
I am supportive of merging with mainline as well. The only proviso is that my experience is with tweaking the Intel codebase and I only have Intel machines to test on, so I am flying blind with Apple Silicon. I do not think I can be of much help in this field. Yes, we should setup a specific fork for testing and preparation, as I do not want to risk disturbing what is already (mostly) working well with our current v17 port.

**Comment by Shambler2** [comment, marked as answer]
So I've noticed that this has actually happened already - https://github.com/sarah-walker-pcem/pcem/pull/14

There's been a few other fixes added as well, though not sure if everything that was patched to run in this version has made it over to mainline. 

Mainline now compiles and runs (I've tested it) on an M1 Mac, though I haven't done any extensive testing to determine if any performance or compatibility issues have popped up.

At the moment its a bit difficult to compile, in fact none of the documentation shows that Mac support is now available, so some updates could still be made to improve the usage on macOS
